tensorflow/compiler/mlir/tensorflow/translate/mlir_roundtrip_flags.cc

```diff
@@ -190,11 +244,54 @@ Status ParseNodeNames(absl::string_view names_str,
   return Status::OK();
 }
 
+StatusOr<std::vector<std::string>> ParseDTypesHelper(
+    absl::string_view data_types_str) {
+  bool inside_subtype = false;
+  int cur_pos = 0;
+  std::vector<std::string> dtypes;
+  for (auto it : llvm::enumerate(data_types_str)) {
+    char c = it.value();
+    int i = it.index();
+    // Skip parsing the subtypes of a type
+    if (c == '(') {
+      if (inside_subtype) {
+        return errors::FailedPrecondition(
+            absl::StrCat("Syntax error in data types '", data_types_str, "'"));
+      }
+      inside_subtype = true;
+    } else if (c == ')') {
+      if (!inside_subtype) {
+        return errors::FailedPrecondition(
+            absl::StrCat("Syntax error in data types '", data_types_str, "'"));
+      }
+      inside_subtype = false;
+    }
+    if (inside_subtype) continue;
+    if (c == ',') {
+      dtypes.push_back(std::string(data_types_str.substr(cur_pos, i)));
+      cur_pos = i + 1;
+    }
+  }
+  if (inside_subtype) {
+    return errors::FailedPrecondition(
+        absl::StrCat("Syntax error in data types '", data_types_str, "'"));
+  }
+  if (!data_types_str.empty()) {
+    dtypes.push_back(
+        std::string(data_types_str.substr(cur_pos, data_types_str.size())));
+  }
+  return dtypes;
+}
+
 Status ParseNodeDataTypes(absl::string_view data_types_str,
                           std::vector<std::string>& data_type_vector) {
   data_type_vector.clear();
   if (!data_types_str.empty()) {
-    data_type_vector = absl::StrSplit(data_types_str, ',');
+    auto s = ParseDTypesHelper(data_types_str);
```

Comment:
TF_ASSIGN_OR_RETURN?

The program shouldn't die if there was a syntax error in the input.